contrib/hashicorp/consul/consul_test.go

```diff
@@ -0,0 +1,249 @@
+package consul
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	consul "github.com/hashicorp/consul/api"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/mocktracer"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMain(m *testing.M) {
+	_, ok := os.LookupEnv("INTEGRATION")
+	if !ok {
+		fmt.Println("--- SKIP: to enable integration test, set the INTEGRATION environment variable")
+		os.Exit(0)
+	}
+	os.Exit(m.Run())
+}
+
+func kv() *KV {
+	client, err := NewClient(consul.DefaultConfig())
+	if err != nil {
+		panic(err)
+	}
+	// Get a handle to the KV API
+	kv := client.KV()
+	return kv
+}
+
+func TestClient(t *testing.T) {
+	assert := assert.New(t)
+	mt := mocktracer.Start()
+	defer mt.Stop()
+
+	client, err := NewClient(consul.DefaultConfig())
+	if err != nil {
+		panic(err)
+	}
+
+	assert.NotNil(client)
+	assert.Nil(err)
+}
+
+func TestClient_error(t *testing.T) {
+	assert := assert.New(t)
+	mt := mocktracer.Start()
+	defer mt.Stop()
+
+	var config = &consul.Config{
+		Address: "badscheme://baduri",
+	}
+
+	client, err := NewClient(config)
+	assert.Nil(client)
+	assert.Error(err)
+
+}
+
+func TestClient_KV(t *testing.T) {
+	assert := assert.New(t)
+	mt := mocktracer.Start()
+	defer mt.Stop()
+
+	client, _ := NewClient(consul.DefaultConfig())
+	clientCtx := client.ctx
+	kv := client.KV()
+	kvCtx := kv.ctx
+
+	assert.Equal(clientCtx, kvCtx)
```

Comment:
It'd be just as clear to get rid of `clientCtx` and `kvCtx`, and assert `(client.ctx, kv.ctx)`